pygmt/clib/session.py

```diff
@@ -1528,21 +1533,29 @@ def virtualfile_from_data(
         ...
         <vector memory>: N = 3 <7/9> <4/6> <1/3>
         """
-        kind = data_kind(data, x, y, z, required_z=required_z)
-
-        if check_kind == "raster" and kind not in ("file", "grid"):
-            raise GMTInvalidInput(f"Unrecognized data type for grid: {type(data)}")
-        if check_kind == "vector" and kind not in (
-            "file",
-            "matrix",
-            "vectors",
-            "geojson",
-        ):
-            raise GMTInvalidInput(f"Unrecognized data type for vector: {type(data)}")
+        kind = data_kind(
+            data, x, y, z, required_z=required_z, required_data=required_data
+        )
+
+        if check_kind == "raster":
+            valid_kinds = ("file", "grid")
+        elif check_kind == "vector":
+            valid_kinds = ("file", "matrix", "vectors", "geojson")
+        else:
+            raise GMTInvalidInput(
+                "Invalid check_kind. Should be either 'raster' or 'vector'."
+            )
```

Comment:
Debating on whether to remove this else argument. Technically this `lib.virtualfile_from_data` function is for internal use only, but the function name is not marked as private with a leading underscore (`_virtualfile_from_data`). Should we assume that others are using this and might get an error with the default `check_kind=None`? Or add a test for `check_kind=None` so that there is code coverage?

Reply:
The docstring says:
```
Used to validate the type of data that can be passed in. Choose     
from 'raster', 'vector', or None. Default is None (no validation).
```
so we should not check the data type if check_kind is None. See 9b31e230fa0f4640e3a0a5bd12f5aad1c58936c0 for the changes.

Reply:
Sounds good. Could you also change the docstring for the expected type at L1489 to `check_kind : str or None` please.

Reply:
Done in 0ed6248bb76bda675c5d87c51b123b48860b7213